Eevee.Sleep.Bot/Modules/SlashCommands/RoleManagementSlashModule.cs

```diff
@@ -0,0 +1,69 @@
+using Discord;
+using Discord.Interactions;
+using JetBrains.Annotations;
+using Eevee.Sleep.Bot.Controllers.Mongo;
+using Eevee.Sleep.Bot.Utils;
+
+namespace Eevee.Sleep.Bot.Modules.SlashCommands;
+
+[Group("role", "Commands for managing the roles on a user.")]
+public class RoleManagementSlashModule : InteractionModuleBase<SocketInteractionContext> {
+    [SlashCommand("change", "Select a specific one from the role you have")]
+    [UsedImplicitly]
+    public Task ChangeRoleAsync() {
+        return Context.Interaction.RespondAsync(
+            components: DiscordMessageMaker.MakeRoleSelectButton(
+                hasRoles: DiscordRoleContoller.FindRoleIdsByUserId(Context.User.Id), 
+                guildRoles: Context.Guild.Roles),
+            ephemeral: true
+        );
+    }
+
+    [SlashCommand("remove", "Remove the role ownership of a user")]
+    [RequireOwner]
+    [UsedImplicitly]
+    public async Task RemoveRoleAsync(IUser user, IRole role) {
+        await DiscordRoleContoller.RemoveRoles(user.Id, [role.Id]);
+
+        await Context.Interaction.RespondAsync(
+            "Role record deleted.",
+            ephemeral: true
+        );
+    }
+
+    [SlashCommand("track", "Track the specific role")]
+    [RequireOwner]
+    [UsedImplicitly]
+    public async Task TrackRoleAsync(IRole role) {
+        await Context.Interaction.RespondAsync(
+            "Role record update/create task dispatched.",
+            ephemeral: true
+        );
+
+        var users = Context.Guild.Users;
```

Comment:
`users` and `roleOwnedUsers` only gets used once, so shouldn't need another variable, unless the line goes too long.